civihr_employee_portal/updates/7044.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Updates contact address, enforcing only one personal address type
+ */
+function civihr_employee_portal_update_7044() {
+  civicrm_initialize();
+  $locationType = civicrm_api3('LocationType', 'get', ['name' => 'Work']);
+  if (!$locationType['count']) {
+    return;
+  }
+
+  $addresses = _get_personal_address_count();
+  foreach ($addresses as $address) {
+    if ($address['total'] != 2) {
```

Comment:
what if the user has 3 or more personal addresses ? this code does not handle that

Reply:
Exactly. I assumed wrongly based on the test scenarios painted. I have pushed an update which catered for any number of primary addresses while also maintaining any existing primary record.